src/commands/createNewProject/NewProjectLanguageStep.ts

```diff
@@ -42,20 +43,29 @@ export class NewProjectLanguageStep extends AzureWizardPromptStep<IProjectWizard
     public async prompt(wizardContext: IProjectWizardContext): Promise<void> {
         const previewDescription: string = localize('previewDescription', '(Preview)');
         // Only display 'supported' languages that can be debugged in VS Code
-        const languagePicks: QuickPickItem[] = [
-            { label: ProjectLanguage.JavaScript },
-            { label: ProjectLanguage.TypeScript },
-            { label: ProjectLanguage.CSharp },
-            { label: ProjectLanguage.Python, description: previewDescription },
-            { label: ProjectLanguage.Java }
+        const languagePicks: IAzureQuickPickItem<ProjectLanguage | undefined>[] = [
+            { label: ProjectLanguage.JavaScript, data: ProjectLanguage.JavaScript },
+            { label: ProjectLanguage.TypeScript, data: ProjectLanguage.TypeScript },
+            { label: ProjectLanguage.CSharp, data: ProjectLanguage.CSharp },
+            { label: ProjectLanguage.Python, description: previewDescription, data: ProjectLanguage.Python },
+            { label: ProjectLanguage.Java, data: ProjectLanguage.Java }
         ];
 
         if (getWorkspaceSetting('enablePowerShell')) {
-            languagePicks.push({ label: ProjectLanguage.PowerShell, description: previewDescription });
+            languagePicks.push({ label: ProjectLanguage.PowerShell, description: previewDescription, data: ProjectLanguage.PowerShell });
         }
 
+        languagePicks.push({ label: localize('viewSamples', '$(link-external) View sample projects'), data: undefined, suppressPersistence: true });
+
         const options: QuickPickOptions = { placeHolder: localize('selectFuncTemplate', 'Select a language for your function project') };
-        wizardContext.language = <ProjectLanguage>(await ext.ui.showQuickPick(languagePicks, options)).label;
+        const result: ProjectLanguage | undefined = (await ext.ui.showQuickPick(languagePicks, options)).data;
+        if (result === undefined) {
+            await openUrl('https://aka.ms/AA4ul9b');
+            wizardContext.actionContext.properties.cancelStep = 'viewSampleProjects';
+            throw new UserCancelledError();
```

Comment:
Didn't we have a custom error wrapper so we didn't have to use the `UserCancelledError`?

Reply:
Uh there's a few things we can do related to `actionContext`, but in this case I think a "cancel" makes sense. Basically they're "cancelling" creating the project in order to view sample projects instead